hack/bin/helm-template.sh

```diff
@@ -4,25 +4,20 @@
 # hack/helm_vars as overrrides, if available.  This allows debugging helm
 # templating issues without actually installing anything, and without needing
 # access to a kubernetes cluster
-USAGE="Usage: $0"
 
 set -e
 
-chart=${1:?$USAGE}
-
 DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 TOP_LEVEL="$DIR/../.."
 CHARTS_DIR="${TOP_LEVEL}/.local/charts"
+: "${FEDERATION_DOMAIN:=example.com}"
+: "${NAMESPACE:=namespace1}"
```

Comment:
I'm not familiar with this syntax. Is that the same as

```
NAMESPACE=${NAMESPACE:-"namespace1"}
```
? i.e. set-to-default-if-not-exists? Or this hardcodes/overrides this variable to always be that?

Reply:
Yes, it's the same as that. `${x:=foo}` sets `$x` to `foo` if not set, and returns the value of `$x`. The colon is just an empty statement. It's a way to evaluate an expression (usually with side effects), without executing any command.

Reply:
Ok, thanks for the explanations!